# Intervalos de confianza

In [50]:
import numpy as np
import scipy.stats as stats
from scipy.stats import t, norm

## Distribución T student 

Para n<30

### Calculandolo manual

In [31]:
n=25
media=158950
S=23800
alfa= 1-0.96 # probabilidad 96%
t=t.ppf(alfa/2,n-1)
t

-2.1715446760080677

In [32]:
ub=media-t*S/np.sqrt(n)
ub

169286.5526577984

In [33]:
lb=media+t*S/np.sqrt(n)
lb

148613.4473422016

### Con función 

In [41]:

confidence_interval = stats.t.interval(0.96,df=25, loc=media, scale=S/np.sqrt(n))
print(confidence_interval)


(148637.0476200049, 169262.9523799951)


### Con un 96% de probabilidad, el valor poblacional estaría entre 148613 y 169286, por lo que no podría ser 180000

## Intervalos de Confianza para diferencias de Medias

In [43]:
# Pilas E
media_e=116.6
s_e=10
n_e=21

In [44]:
# Pilas D
media_d=115.8
s_d=15
n_d=n_e

In [45]:
#Probabilicad del 97%
alfa=1-0.97

In [47]:
import scipy.stats as stats

# Valores conocidos para la primera muestra
media1 = media_e  # Media de la primera población
desviacion_estandar1 = s_e  # Desviación estándar de la primera población
n1 = n_e  # Tamaño de la primera muestra

# Valores conocidos para la segunda muestra
media2 = media_d  # Media de la segunda población
desviacion_estandar2 = s_d  # Desviación estándar de la segunda población
n2 = n_d  # Tamaño de la segunda muestra

nivel_confianza = 0.97  # Nivel de confianza del 97%

# Calcular el error estándar de la diferencia de medias
sem_diff = ((desviacion_estandar1 ** 2 / n1) + (desviacion_estandar2 ** 2 / n2)) ** 0.5

# Calcular el intervalo de confianza para la diferencia de medias
intervalo = stats.t.interval(nivel_confianza, loc=media1 - media2, scale=sem_diff, df=min(n1 - 1, n2 - 1))

print("Intervalo de confianza para la diferencia de medias:", intervalo)


Intervalo de confianza para la diferencia de medias: (-8.39072750721638, 9.990727507216373)


## Intervalos de confianza para la proporción (distribuciones Binomiales)

In [49]:
n=55
X=11 # Casos de fallo a 250°F
alfa=1-0.9 #Probabilidad del 90%
p=X/n

# ¿se puede afirmar que con una confianza del 90% no hay casos de fallo a 250°F?

In [51]:
p-abs(norm.ppf(alfa/2))*np.sqrt(p*(1-p)/n)

0.11128319287497217

In [52]:
p+abs(norm.ppf(alfa/2))*np.sqrt(p*(1-p)/n)

0.28871680712502784

#### El IC es [0.11 , 0.288]

Dado que el intervalo tiene 28.8% no es cierta la afirmaicón de que el 80% de las mascaras soporta, es decir, el valor máximo del intervalo debería ser 0.2